src/extension.ts

```diff
@@ -99,72 +117,72 @@ export function deactivate() {
-    let folder = folders[0];
-    if (folder) {
-      executablePath = executablePath
-        .replace('${workspaceFolder}', folder.uri.path)
-        .replace('${workspaceRoot}', folder.uri.path);
+// Given a path to executable that contains common vscode-specific path variables,
+// replaces those variables with proper values.
+function interpolateVSCodeSpecificExecutablePath(executablePath: string): string {
+    // Interpolate in values for common vscode path variables
+
+    // Path variable for home directory
+    executablePath = executablePath
+        .replace('${HOME}', os.homedir)
+        .replace('${home}', os.homedir)
+        .replace(/^~/, os.homedir);
+
+    // Path variable for workspace folder
+    let folders = workspace.workspaceFolders;
+    if (folders) {
+        let folder = folders[0];
+        if (folder) {
+            executablePath = executablePath
+                .replace('${workspaceFolder}', folder.uri.path)
+                .replace('${workspaceRoot}', folder.uri.path);
+        }
     }
-  }
 
-  console.log(`Location after path variables subsitution: ${executablePath}`);
-
-  return {
-    executablePath,
-    usePathWaspls: false,
-  };
+    return executablePath
 }
 
-// Runs the executable in `version` mode and checks that it:
+// Runs the wasp executable in `version` mode and checks that it:
 // - Exits successfuly
 // - Exits within 1 second
 //
 // If it fails either of these checks, a status is returned to show that the
 // executable path is not valid.
-async function checkExecutable(executablePath: string): Promise<Status> {
-  const execPromise = promisify(execFile)
-    (executablePath, ['version'], { timeout: 2000, windowsHide: true })
-    .then(() => Status.Available).catch(_err => Status.Missing);
-
-  const timeoutPromise = new Promise<Status>((resolve, _reject) => {
-    let timer = setTimeout(() => {
-      clearTimeout(timer);
-      resolve(Status.Timedout);
-    }, 1000);
-  });
+async function getWaspExecutableVersion(executablePath: string): Promise<ExecutableErrorStatus | string> {
+  const execFileP = promisify(execFile);
+  const execPromise = execFileP(executablePath, ['version'], { timeout: 2000, windowsHide: true })
+    .then(({stdout}) => stdout)
+    .catch(_err => ExecutableErrorStatus.Missing);
+
+  const timeoutPromise = new Promise<ExecutableErrorStatus>((resolve) => setTimeout(() => resolve(ExecutableErrorStatus.Timedout), 1000));
 
   return Promise.race([execPromise, timeoutPromise]);
 }
 
-// Executable status types
-enum Status {
-  Available = 'available',
-  Missing = 'missing',
-  Timedout = 'timedout',
+enum ExecutableErrorStatus { Missing, Timedout }
+
+// Given two semantic version strings, returns -1 if first version is smaller than the second,
+// 0 if they are the same, or 1 if second version is bigger than the first.
+function compareSimpleSemvers(v1: string, v2: string) {
+  function parseSemver(v:string) { return v.split('.').map(n => parseInt(n)) };
```

Comment:
Not really a performance issue (well technically it can be, but definitely not in this case), but global functions are not really a problem (i.e., global functions are considered default). Idiomatic JavaScript generally uses nested function definitions when they're needed for scoping purposes.

Seeing JavaScript function that's nested just because it's short and only used locally is a bit strange (I immediately start looking for a "catch"), not due to any objective reasons (at least most of the time), just a convention thing.

Reply:
So maybe it is ok to have it! THere is a reason -> I want to use it only here, and I am not keen to others using it. Kind of like `where` in Haskell, as you said.

Reply:
Ok, I see you really like it, we can leave it as it is :smile: 
If we are keeping it in though, can you just space it into three rows to make it less unusual (or use an arrow)?

Now I'll go into another rant trying to explain my reasoning in case I wasn't clear enough. But again, if I was clear and you have all the info, but you simply prefer it like this - no problem, it stays and we can end the discussion.

----

It's unusual because most people with extensive JS experience are going to be thrown off by seeing a nested function. It's a legitimate construct, but not a construct that's often used in JavaScript - it's just a question of idiomatic code. Something like expecting `filter even list` instead of `[x | x <- list, even x]` in Haskell.

When one wants encapsulation/privacy in JS, they generally create a new module and only export what you want to export.

Reply:
I actually moved it out for your sake, but I commented I actually prefer leaving it in.
I get the arguments, I just don't find them strong enough, as they are mostly based on "tradition" vs what I think brings additional meaning (it can be used only inside because author wanted it to be used only inside).

Reply:
Haha, we sure spent a lot of time discussing where this function will end up and it's mostly my fault, sorry :sweat_smile: